sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -185,8 +183,21 @@ case class DataSource(
     }
 
     SchemaUtils.checkColumnNameDuplication(
```

Comment:
I checked and I found we needed to leave the check for partition schema because `InMemoryFileIndex` first enumerates partitions inside this function. So, If we remove this check, the tests fail in https://github.com/apache/spark/pull/17758/files#diff-b9ddfbc9be8d83ecf100b3b8ff9610b9R721